mne_bids/write.py

```diff
@@ -911,13 +911,17 @@ def make_bids_basename(subject=None, session=None, task=None,
         raise ValueError("At least one parameter must be given.")
 
     if subject == 'emptyroom':
-        if suffix is None and task != 'noise':
-            raise ValueError('task must be ''noise'' if subject is'
-                             ' ''emptyroom''')
+        if task != 'noise':
+            msg = f"empty-room task must be 'noise', but received: {task}"
+            raise ValueError(msg)
+
         try:
             datetime.strptime(session, '%Y%m%d')
-        except ValueError:
-            raise ValueError("session must be string of format YYYYMMDD")
+        except (TypeError, ValueError):
+            msg = (f'empty-room session should be a string of format '
+                   f'YYYYMMDD, but received: {session}. We trust that you '
+                   f'know what you are doing.')
```

Comment:
lol ... trust the user :) any particular reason to support this? I would actually suggest leaving it like before so that we make people move towards more standardized file structures. It's good for the community.

Reply:
Now this is a tough one, I ran into this because when we do `get_matched_empty_room()`, we will have to handle empty-room recordings that were _not_ saved with a `ses-YYYYMMDD` attribute (but, e.g., `ses-01`). Now to read these files (or any files, for that matter) via `read_raw_bids`, I first need to generate their `bids_basename` – and in `master`, it will raise an exception. 

A possible solution could be to factor out most of `read_raw_bids()` into a private function that doesn't do this kind of sanity check (or allows us to override it), and call this private function from `get_matched_empty_room()` to generate the basename.

Reply:
yes I like the solution of factoring out. Since it's a SHOULD :) :) :) we _should_ try to help and direct the user as much as possible to do things in a BIDS way

Reply:
> Since it's a SHOULD :) :) :)

💩💩💩

![68747470733a2f2f68756e67617279746f6461792e68752f77702d636f6e74656e742f75706c6f6164732f323031382f30322f3138707332372e6a7067](https://user-images.githubusercontent.com/2046265/82591042-fe314000-9b9e-11ea-98af-828fa3d3d7d0.jpg)

Reply:
Added this.